persistence/modeshape-persistence-relational/src/main/java/org/modeshape/persistence/relational/StatementsProvider.java

```diff
@@ -104,6 +108,26 @@ protected Void dropTable(Connection connection) throws SQLException {
         }
     }
 
+    protected <T> T load(Connection connection, Set<String> ids, Function<ResultSet, T> function) throws SQLException {
+        logDebug("Loading ids {0} from {1}", ids.toString(), tableName);
+        String params = ids.stream().map(id -> "?").collect(Collectors.joining(","));
+        String statementString = statements.get(GET_MULTIPLE).replaceAll("#", params);
+        try (PreparedStatement ps = connection.prepareStatement(statementString)) {
+            AtomicInteger counter = new AtomicInteger(1);
+            ids.forEach(id -> {
+                try {
+                    ps.setString(counter.getAndIncrement(), id);
+                } catch (SQLException e) {
+                    throw new RelationalProviderException(e);
+                }
+            });
+            try (ResultSet rs = ps.executeQuery()) {
+                return function.apply(rs);
+            }
+        }
+    }
+    
+
```

Comment:
* [MySQL](http://dev.mysql.com/doc/refman/5.1/en/comparison-operators.html#function_in): "limited by the max_allowed_packet value."
* [SQL Server](https://msdn.microsoft.com/en-us/library/ms143432.aspx): "limited by 65,536 * Network Packet Size" (which defaults to 4K).
* [Oracle](http://stackoverflow.com/questions/19002792/why-oracle-in-clause-has-limit-of-1000-only-for-static-data) has an expression limit of 1000 values.
* PostgreSQL [apparently](http://stackoverflow.com/questions/2426331/in-mysql-or-postgres-is-there-a-limit-to-the-size-of-an-in-1-2-n-statement) slows dramatically with lots of values.

Perhaps you could add a warnings to the JavaDoc and/or log stating that the method may not work with large numbers of IDs.